core/src/main/kotlin/org/opensearch/alerting/core/lock/LockService.kt

```diff
@@ -72,13 +75,23 @@ class LockService(private val client: Client, private val clusterService: Cluste
                                 object : ActionListener<LockModel> {
                                     override fun onResponse(existingLock: LockModel?) {
                                         if (existingLock != null) {
+                                            val currentTimestamp = getNow()
                                             if (isLockReleased(existingLock)) {
                                                 log.debug("lock is released or expired: {}", existingLock)
                                                 val updateLock = LockModel(existingLock, getNow(), false)
                                                 updateLock(updateLock, listener)
                                             } else {
-                                                log.debug("Lock is NOT released or expired. {}", existingLock)
-                                                listener.onResponse(null)
+                                                log.debug("Lock is NOT released. {}", existingLock)
+                                                if (existingLock.lockTime.epochSecond + LOCK_EXPIRED_SECONDS.seconds
+                                                    < currentTimestamp.epochSecond
+                                                ) {
+                                                    log.debug("Lock is expired. Renewing Lock {}", existingLock)
+                                                    val updateLock = LockModel(existingLock, getNow(), false)
+                                                    updateLock(updateLock, listener)
```

Comment:
What happens if the previous node that owned the lock resumes its execution? i.e. node holding lock was unresponsive due to high traffic and temporarily left the cluster. After traffic subsides it rejoins the cluster

Reply:
Related question - are there any race conditions with multiple nodes attempting to claim an expired lock?

Reply:
hi @engechas , each lock is specific to a monitor. & if the node running the old job goes down, the job is cancelled. the monitor job is then reassigned to a new node & it can then try to renew the lock with this fix now.